controllers/slicegateway/slicegateway.go

```diff
@@ -1377,16 +1358,32 @@ func (r *SliceGwReconciler) ReconcileGatewayDeployments(ctx context.Context, sli
 
 	// Reconcile deployment to node port mapping for gw client deployments
 	if isClient(sliceGw) {
+		UpdateDeploymentSlice := make([]appsv1.Deployment, 0)
+		toUpdate := false
 		for _, deployment := range deployments.Items {
 			found, nodePortInUse := getClientGwRemotePortInUse(ctx, r.Client, sliceGw, deployment.Name)
 			if found {
```

Comment:
I think this block can be simplified and made more readable if we first check if the port in use is actually present in the list of remote port numbers in the slicegw object. 

```
for _, deployment := range deployments.Items {
			found, nodePortInUse := getClientGwRemotePortInUse(ctx, r.Client, sliceGw, deployment.Name)
			if found {
                            // Check if the portInUse is valid. 
                            // It is valid only if the list of remoteNodePorts in the slicegw object contains the portInUse.
                            if !checkIfNodePortIsValid(sliceGw.Status.Config.SliceGatewayRemoteNodePorts, 
                                                                       nodePortInUse) {
                                // Get a valid port number for this deployment
                                portNumToUpdate := 
                                    allocateNodePortToClient(sliceGw.Status.Config.SliceGatewayRemoteNodePorts,     
                                                                                &gwClientToRemotePortMap)
                                // Update the port map
                                gwClientToRemotePortMap.Store(deployment.Name, portNumToUpdate)
                                err := r.updateGatewayDeploymentNodePort(ctx, r.Client, sliceGw, &deployment, portNumToUpdate)
                                if err != nil {
                                    return ctrl.Result{}, err, true
                                }
                                // Requeue if the update was successful
                                return ctrl.Result{}, nil, true
                          }
                          // At this point, the node port in use is valid. Check if the port map is populated. Only case
                         // where it is not populated is if the operator restarts. The populated value must match the
                         // port in use. If not, the deploy needs to be updated to match the state stored in the operator.
                         portInMap, foundInMap := gwClientToRemotePortMap.Load(deployment.Name) 
                         if foundInMap {
                             if portInMap != nodePortInUse {
                                 // Update the deployment since the port numbers do not match
                                 err := r.updateGatewayDeploymentNodePort(ctx, r.Client, sliceGw, &deployment, portInMap)
                                if err != nil {
                                    return ctrl.Result{}, err, true
                                }
                                // Requeue if the update was successful
                                return ctrl.Result{}, nil, true
                         } else {
                             gwClientToRemotePortMap.Store(deployment.Name, nodePortInUse)
                         }
               }
}
```

Note: checkIfNodePortIsValid() is already defined in the utils.go file.

Reply:
@bharath-avesha thanks for the review. I initially went with the same approach but there was a weird case where the nodePorts in the slicegw config have same values (this happens only once or twice after reconciler recognises the change in nodeports from the controller side). When this happens the the values in the map are altered (both the deployments having the same values). So for this reason in the first loop i am making sure that the map has correct values and then updating the deployments(if any) in the second. I will try to reproduce the issue and update the PR accordingly.

Reply:
Thanks for the update @kon3m.. Lets try this change and see if the duplicate port number issue comes up.. Plz collect the operator logs and let me know..

Reply:
@bharath-avesha i have tried to reproduce it yesterday and it did show up again and i wanted to record/screencast the scenario but it is not consistent and was unable to capture/reproduce it again. Can i just apply the above change with a line before the loop like:

`if hasSameValues(sliceGw.Status.Config.SliceGatewayRemoteNodePorts) {
    return ctrl.Result{}, nil, true
 }
` 
just requeue if it has same port numbers.

Reply:
@bharath-avesha i have tried reproducing the scenario after applying the above change , the issue doesn't seem to show up. Updated the PR with the change.